packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -190,48 +164,36 @@ class TransactionService {
       if (perfOptions.pageLoadSampled) {
         tr.sampled = perfOptions.pageLoadSampled
       }
-
-      if (tr.name === 'Unknown' && perfOptions.pageLoadTransactionName) {
```

Comment:
Shouldn't we set the transaction name here? I think we should set the transaction name as early as possible, I know we also try to set the name onEnd but they are both necessary.

Reply:
Yeah I felt there was no reason to do the same thing twice, I cannot think of any scenario where we might fail to update the page load transaction name. Can you think of a scenario where this would be valid?

Reply:
For example if the users try to get the current transaction based before the transaction has ended, they would see the wrong name. Also if we decide to influence sampling based on the transaction name (e.g. sampling higher rate for important transactions) since we make the sampling decision at the start of transaction we should also set the name.

Reply:
Hmmm, yeah sounds good. I will just add a comment in code mentioning the same.